### PR TITLE
Plugin not found

### PR DESCRIPTION
**Pull request by vogdb**
Cannot find method `processMessage` of undefined which in this case should be `Plugin`. Plugin is accesed via `return this.getPlugin(pluginId).processMessage(message);`
Log:
```
app debug <- janus { janus: 'hangup',
  session_id: 3280711602,
  sender: 2864469971,
  reason: 'DTLS alert' }
app error [TypeError: Cannot read property 'processMessage' of undefined] TypeError: Cannot read property 'processMessage' of undefined
    at ProxyConnection.processMessage (/Users/tomasz/projects/cargomedia/cm-janus/lib/proxy-connection.js:27:36)
    at JanusConnection.<anonymous> (/Users/tomasz/projects/cargomedia/cm-janus/lib/janus-proxy.js:42:23)
    at JanusConnection.emit (events.js:107:17)
    at JanusConnection.Connection._onMessage (/Users/tomasz/projects/cargomedia/cm-janus/lib/connection.js:78:10)
    at JanusConnection.<anonymous> (/Users/tomasz/projects/cargomedia/cm-janus/lib/connection.js:18:10)
    at WebSocket.emit (events.js:110:17)
    at Receiver.ontext (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/WebSocket.js:816:10)
    at /Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:477:18
    at Receiver.applyExtensions (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:364:5)
    at /Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:466:14
    at Receiver.flush (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:340:3)
    at Receiver.opcodes.1.finish (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:482:12)
    at Receiver.expectHandler (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:457:31)
    at Receiver.add (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:95:24)
    at Socket.realHandler (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/WebSocket.js:800:20)
    at Socket.emit (events.js:107:17)
    at readableAddChunk (_stream_readable.js:163:16)
    at Socket.Readable.push (_stream_readable.js:126:10)
    at TCP.onread (net.js:538:20)
app debug <- janus { janus: 'success',
  session_id: 3280711602,
  transaction: 'TInwqbQr8zj8' }
app debug <- janus { janus: 'detached', sender: 2864469971 }
app error [TypeError: Cannot read property 'processMessage' of undefined] TypeError: Cannot read property 'processMessage' of undefined
    at ProxyConnection.processMessage (/Users/tomasz/projects/cargomedia/cm-janus/lib/proxy-connection.js:27:36)
    at JanusConnection.<anonymous> (/Users/tomasz/projects/cargomedia/cm-janus/lib/janus-proxy.js:42:23)
    at JanusConnection.emit (events.js:107:17)
    at JanusConnection.Connection._onMessage (/Users/tomasz/projects/cargomedia/cm-janus/lib/connection.js:78:10)
    at JanusConnection.<anonymous> (/Users/tomasz/projects/cargomedia/cm-janus/lib/connection.js:18:10)
    at WebSocket.emit (events.js:110:17)
    at Receiver.ontext (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/WebSocket.js:816:10)
    at /Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:477:18
    at Receiver.applyExtensions (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:364:5)
    at /Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:466:14
    at Receiver.flush (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:340:3)
    at Receiver.opcodes.1.finish (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:482:12)
    at Receiver.expectHandler (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:457:31)
    at Receiver.add (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/Receiver.js:95:24)
    at Socket.realHandler (/Users/tomasz/projects/cargomedia/cm-janus/node_modules/ws/lib/WebSocket.js:800:20)
    at Socket.emit (events.js:107:17)
    at readableAddChunk (_stream_readable.js:163:16)
    at Socket.Readable.push (_stream_readable.js:126:10)
    at TCP.onread (net.js:538:20)
```

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review.

File: lib/connection.js [hunk 1]
```diff
@@ -34,13 +34,17 @@ Connection.prototype.close = function() {
...
 Connection.prototype.send = function(message) {
-  if (WebSocket.OPEN === this.webSocket.readyState) {
+  if (this.isOpened()) {
     return this._send(message);
   } else {
     return this._queue(message);
   }
 };
 
+Connection.prototype.isOpened = function() {
```
Comment (tomaszdurka) [inline lib/connection.js]: Add docu :)

File: lib/proxy-connection.js [hunk 2]
```diff
@@ -58,6 +61,22 @@ ProxyConnection.prototype.onAttach = function(message) {
...
+ProxyConnection.prototype.destroy = function() {
+  if (this.browserConnection.isOpened()) {
+    this.browserConnection.removeAllListeners();
+    this.browserConnection.close();
+  }
+  if (this.janusConnection.isOpened()) {
+    this.janusConnection.removeAllListeners();
+    this.janusConnection.close();
+  }
+  streams.findAllByConnection(this.browserConnection).forEach(function(stream) {
```
Comment (tomaszdurka) [inline lib/proxy-connection.js]: Make sure to use `proxyConnection` for each `streams` usage.

**Comment by tomaszdurka** [comment]
Rest looks good.

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@tomaszdurka wdyt about `ProxyConnection` to `ProxyBridge`, `TupleConnection`, `ConnectionPair`?

File: lib/proxy-connection.js [hunk 3]
```diff
@@ -58,6 +61,22 @@ ProxyConnection.prototype.onAttach = function(message) {
 };
 
 
+ProxyConnection.prototype.close = function() {
+  if (this.browserConnection.isOpened()) {
+    this.browserConnection.removeAllListeners('message');
+    this.browserConnection.close();
```
Comment (tomaszdurka) [inline lib/proxy-connection.js]: What happens if you close already closed connection?

Comment (tomaszdurka) [inline lib/proxy-connection.js]: Oh, didn't notice check about isOpen before.

**Comment by tomaszdurka** [comment]
Looks good.

**Pull request merged**